plugin/ui/completion.py

```diff
@@ -274,3 +298,95 @@ def show(self) -> None:
     @staticmethod
     def hide(view: sublime.View) -> None:
         mdpopups.hide_popup(view)
+
+
+class _PhantomCompletion(_BaseCompletion):
+    COPILOT_PHANTOM_COMPLETION = "copilot_phantom_completion"
+    PHANTOM_TEMPLATE = """
+    <body id="copilot-completion">
+        <style>
+            body {{
+                color: #808080;
+            }}
+
+            .copilot-completion-line {{
+                margin-top: {line_padding_top}px;
+                margin-bottom: {line_padding_bottom}px;
+            }}
+
+            .copilot-completion-line.first {{
+                margin-top: 0;
+            }}
+        </style>
+        {body}
+    </body>
+    """
+    PHANTOM_LINE_TEMPLATE = '<div class="copilot-completion-line {class_name}">{content}</div>'
+
+    def __init__(self, view: sublime.View) -> None:
+        super().__init__(view)
+
+        self._phantom_set = self._get_phantom_set(view)
+
+    @classmethod
+    def _get_phantom_set(cls, view: sublime.View) -> sublime.PhantomSet:
+        view_id = view.id()
+
+        # create phantom set if there is no existing one
+        if not _view_to_phantom_set.get(view_id):
+            _view_to_phantom_set[view_id] = sublime.PhantomSet(view, cls.COPILOT_PHANTOM_COMPLETION)
+
+        return _view_to_phantom_set[view_id]
+
+    def normalize_phantom_line(self, line: str) -> str:
+        return html.escape(line).replace(" ", "&nbsp;").replace("\t", "&nbsp;" * self._settings.get("tab_size"))
+
+    def _build_phantom(
+        self,
+        lines: Union[str, Sequence[str]],
+        begin: int,
+        end: Optional[int] = None,
+        *,
+        inline: bool = True
+        # format separator
+    ) -> sublime.Phantom:
+        body = (
+            self.normalize_phantom_line(lines)
+            if isinstance(lines, str)
+            else "".join(
+                self.PHANTOM_LINE_TEMPLATE.format(
+                    class_name=("rest" if index else "first"),
+                    content=self.normalize_phantom_line(line),
+                )
+                for index, line in enumerate(lines)
+            )
+        )
+
+        return sublime.Phantom(
+            sublime.Region(begin, begin if end is None else end),
+            self.PHANTOM_TEMPLATE.format(
+                body=body,
+                line_padding_top=int(self._settings.get("line_padding_top")) * 2,  # TODO: play with this more
+                line_padding_bottom=int(self._settings.get("line_padding_bottom")) * 2,
+            ),
+            sublime.LAYOUT_INLINE if inline else sublime.LAYOUT_BLOCK,
+        )
+
+    def show(self) -> None:
+        completion = self.completion_manager.current_completion
+        assert completion
+
+        first_line, *rest_lines = completion["displayText"].splitlines()
+
+        assert self._phantom_set
+        self._phantom_set.update(
+            [
+                self._build_phantom([first_line], completion["point"] + 1, completion["point"]),
```

Comment:
```suggestion
                self._build_phantom(first_line, completion["point"] + 1, completion["point"]),
```

Just notice that I forgot to remove the brackets.